docs: Admin upgrade guide (#3621)
medusajs/medusa: docs/content/upgrade-guides/admin/1-0-0.md

> # Medusa Admin: 1.0.0

Reviewer: ```suggestion
---
description: 'Migrate from the GitHub repository to the NPM package.'
---

# Medusa Admin: 1.0.0
```

> Medusa Admin has been moved from a standalone repository to our mono-repository to centralise projects and ship it as an NPM package.

Reviewer: suggestion: is it important to highlight that it was moved to our mono-repository? If you think it is, then I'm fine with keeping it. If not, I suggest just focusing that it's now an NPM package that can be installed on the backend. Something like this:

```suggestion
Medusa Admin has been moved from being hosted on a GitHub repository to being published as an NPM package that can be installed in the same project as the Medusa backend. This upgrade guide will help you find the right resources to make the upgrade.
```

## Overview

> The NPM package name is `@medusajs/admin-ui` and in our core repository the UI can be found within `packages/admin-ui`. 

Reviewer: Some comment as the previous suggestion. If it's not important to highlight details regarding the repository, then I think we can just say that admin is now composed of two packages `@medusajs/admin` and `@medusajs/admin-ui`, and explaining generally what each is.

If it's important to highlight the repository, then we should use the same terms when referring to it. Before you mentioned it as "mono-repository" but now it's "core repository"

This guide will cover three cases where the move might affect your current setup:

- Updating your existing project to use the plugin
- Replacing your already deployed admin with the plugin
- Preserving customizations in the plugin

## Actions required

> **Updating your existing project to use the plugin**

Reviewer: ```suggestion
### Updating your existing project to use the plugin
```

As mentioned, Medusa Admin is now distributed via NPM and installed as a plugins. To replace your existing admin, you first need to install the plugin in your Medusa server project.

> Follow our quickstart guide [INSERT LINK] to get started setting up the plugin.

Reviewer: ```suggestion
Follow the [admin quickstart guide](../../admin/quickstart.mdx) to learn how to set up the plugin with the Medusa backend.
```

> **Replacing your already deployed admin with the plugin**

Reviewer: ```suggestion
### Deploy the New Admin Plugin
```

The move to an NPM package has implications for the workflow to deploy the admin. You can still use a platform like Vercel or Netlify to host the project, but that requires you to eject the admin system from the plugin instead of serving it on the server.

> 

Reviewer 2: I think the wording might be a bit confusing here. There is no need to eject the admin when hosting on Vercel etc.

> Refer to our guide on deploying to Vercel [INSERT LINK] to see an example of the new workflow. The same steps apply to most hosting platforms. 

Reviewer: ```suggestion
You can learn about the new deployment process through the [Vercel Deployment Guide](../../deployment/admin/deploying-on-vercel.md). The process should be similar for other hosting platforms as well.
```

> **Preserving customizations in the plugin**

Reviewer: ```suggestion
### Preserving Customizations in the Admin
```

> To keep your custom components or general changes to Medusa Admin, you have to eject the admin system from the plugin instead of serving it on the server.

Reviewer 2: I think this section should be something along the lines of: 

There is currently no seamless way to migrate any custom components or general changes to Medusa Admin to the new plugin. We are actively working on supporting Admin extensions, and expect to ship the feature within the next quarter. In the meantime, if you want to keep up with new features that are added to Admin, you will need to use `medusa-admin eject` to access the source code, and can then compare your Admin to the ejected code and migrate any changes between the two. We will be maintaining the current Admin repository for the foreseeable future, so if you don't plan to use any new features added to Medusa, you can continue using the current repository to pull bug fixes, etc.

This guide will cover three cases where the move might affect your current setup:

- Updating your existing project to use the plugin
- Replacing your already deployed admin with the plugin
- Preserving customizations in the plugin

## Actions required

### Updating your existing project to use the plugin

As mentioned, Medusa Admin is now distributed via NPM and installed as a plugins. To replace your existing admin, you first need to install the plugin in your Medusa server project.

Follow the [admin quickstart guide](../../admin/quickstart.mdx) to learn how to set up the plugin with the Medusa backend.

### Deploy the New Admin Plugin

The move to an NPM package has implications for the workflow to deploy the Admin. The plugin offers to ways to deploy the admin, either by serving the Admin directly from the Medusa server, or by deploying the Admin to a separate hosting platform.

To host the Admin directly from your Medusa server, you can follow the [Admin Quickstart Guide](../../admin/quickstart.mdx).

> You can learn how to deploy the Admin to a host through the [Vercel Deployment Guide](../../deployment/admin/deploying-on-vercel.md). The process is similar for other Git based hosting platforms.

Reviewer: Sorry, typo 😅

```suggestion
You can learn how to deploy the Admin to a host through the [Vercel Deployment Guide](../../deployments/admin/deploying-on-vercel.md). The process is similar for other Git based hosting platforms.
```

### Preserving Customizations in the Admin

> If you have made customizations to the Admin, it is recommend that you keep your current setup and wait for the next minor release of Medusa Admin, that will introduce a more seamless way to extend the UI. Migrating customizations to the new admin plugin is not supported, and will require manaully patching any changes or forking the Admin plugin. The current Admin repository will still receive critical bug fixes while in maintenance, but it will not be updated with new features. If you can forego the latest features introduced in 1.8 you can continue to use the standalone admin repository for now.

Reviewer: suggestion: since we are recommending them to wait before updating, maybe it makes more sense to add a note about it before the "Overview" saying something like:

```
:::note

If you made customizations to the admin and you want to preserve these customization, then it's recommended not to update to use the new admin plugin. You can learn more in [this section](#preserving-customizations-in-the-admin).

:::
```

> If you want to keep your customizations while also upgrading to the latest version of Medusa Admin, you have two options:

Reviewer: we should indicate here that these options are not recommended. Otherwise it might seem confusing that we're saying don't update, then saying there are options to mitigate this issue.
